rtos/source/ThisThread.cpp

```diff
@@ -251,4 +251,32 @@ const char *get_name()
 #endif
 }
 
+osPriority_t ThisThread::get_priority()
+{
+#if MBED_CONF_RTOS_PRESENT
+    osThreadId_t id = osThreadGetId();
+    if (id == nullptr) {
+        return osPriorityError;
+    }
+
+    return osThreadGetPriority(id);
+#else
+    return (osPriorityError)(-1);
+#endif
+}
+
+osStatus_t ThisThread::suspend()
+{
+#if MBED_CONF_RTOS_PRESENT
+    osThreadId_t id = osThreadGetId();
+    if (id == nullptr) {
+        return osError;
+    }
+
+    return osThreadSuspend(id);
+#else
+    return (osError)(-1);
```

Comment:
The definition of `osError ` might not be available if `MBED_CONF_RTOS_PRESENT ` is not present.

Reply:
Fixed

Reply:
Can we try a baremetal build and see what breaks? Will CI already try for us?

Reply:
Did try that, it doesn't compile ThisThread.cpp, Semaphore.cpp or Mutex.cpp

Reply:
You might need to attempt to use a Semaphore in order to properly test. Otherwise, what would be the point of the ifdefs in those classes... There must be a way to get those classes included.

Reply:
yes, tried a different method, the way to include is changing `"requires": ["bare-metal","rtos-api"]` in mbed_app.json on https://github.com/armmbed/mbed-os-example-blinky-baremetal/
Fixes to follow soon.